### PR TITLE
Implemented withCssHandles higher order component

### PR DESCRIPTION
**Pull request by iaronaraujo**
#### What does this PR do? \*
Create a `withCssHandles` higher-order component to be used in cases that you can't use the `useCssHandles`

#### How to test it? \*
Go to the react folder and run `yarn test`

### CODE REVIEW

**Comment by victorhmp** [comment]
🎉

File: docs/README.md [hunk 1]
```diff
@@ -36,13 +39,51 @@ const Component: FunctionComponent = () => {
 }
 ```
 
+### withCssHandles
+
+```tsx
+import React, { Component } from 'react'
```
Comment (lbebber) [inline docs/README.md]: make this example shorter, only showing basic functionality, not every cssHandles usage case

File: react/withCssHandles.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+const withCssHandles = <T extends CssHandlesInput, ComponentProps>(
+  handles: T,
+  options?: CssHandlesOptions
+) => (Component: ComponentType<ComponentProps & { cssHandles: CssHandles<T> }>) => {
+  const EnhancedComponent = (props: ComponentProps) => {
+    const cssHandles = useCssHandles(handles, options)
+
+    return <Component cssHandles={cssHandles} {...props} />
+  }
```
Comment (lbebber) [inline react/withCssHandles.tsx]:
```suggestion
  }
  
  const displayName = Component.displayName || Component.name || 'Component'

  EnhancedComponent.displayName = `withCssHandles(${displayName})`
```

File: docs/README.md [hunk 3]
```diff
@@ -36,13 +40,42 @@ const Component: FunctionComponent = () => {
...
+import { withCssHandles } from from 'vtex.css-handles'
+
+const CSS_HANDLES = ['container', 'background', 'text'] as const
+
+class ExampleComponent extends Component {
+
+  render() {
+    const { cssHandles } = this.props
+
+    return (
```
Comment (lbebber) [inline docs/README.md]: make it even shorter pls 🙏  one line is enough, the focus is the last line

**Comment by lbebber** [comment]
Force-merging as an admin because the docs bot is stuck

**Pull request merged**